### ISSUE TITLE
Optimizely breaks Gatsby

### ISSUE THREAD

**Issue by moshie** [optimizely/react-sdk#25]
When attempting a build in Gatsby optimizely throws several es export / import errors

FYI I used this as an example: https://github.com/circAssimilate/gatsby-with-optimizely-full-stack

```
$ gatsby build
success open and validate gatsby-configs - 0.009s
success load plugins - 0.110s
success onPreInit - 0.004s
success delete html and css files from previous builds - 0.008s
success initialize cache - 0.006s
success copy gatsby files - 0.049s
success onPreBootstrap - 0.018s
success createSchemaCustomization - 0.002s
success source and transform nodes - 0.038s
success building schema - 0.136s
success createPages - 0.305s
success createPagesStatefully - 0.021s
success onPreExtractQueries - 0.002s
warn There are conflicting field types in your data.

If you have explicitly defined a type for those fields, you can safely ignore this warning message.
Otherwise, Gatsby will omit those fields from the GraphQL schema.

If you know all field types in advance, the best strategy is to explicitly define them with the `createTypes` action, and skip inference with
 the `@dontInfer` directive.
SitePage.context.datafile.typedAudiences.conditions:
 - type: [string,array]
   value: [ 'and', [  ] ]
success update schema - 0.094s
success extract queries from components - 0.021s
success write out requires - 0.009s
success write out redirect data - 0.002s
success onPostBootstrap - 0.001s
⠀
info bootstrap finished - 3.066 s
⠀
success Building production JavaScript and CSS bundles - 7.723s
success Rewriting compilation hashes - 0.006s
success run queries - 7.869s - 1/1 0.13/s

 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'Children' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'Component' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'Component' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'Component' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'Component' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'Component' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'cloneElement' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'createContext' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'createElement' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'createElement' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'createElement' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'createElement' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'createInstance' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'forwardRef' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'getLogger' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'getLogger' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'getLogger' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'getLogger' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'getLogger' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'isValidElement' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't reexport the named export 'enums' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't reexport the named export 'errorHandler' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't reexport the named export 'eventDispatcher' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't reexport the named export 'logging' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't reexport the named export 'setLogLevel' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't reexport the named export 'setLogger' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs

not finished Building static HTML for pages - 1.094s
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by timclark97** [comment]
also experiencing this issue with SSR.

**Comment by moshie** [comment]
Thanks @timclark97 that looks to be working now!

**Comment by samblackk** [comment]
Doesn't seem like their including `react-sdk.browser.umd` in their build anymore. Alternative? Running into this as well.

**Comment by jderochervlk** [comment]
@samblackk You have to use v1 of the SDK instead of the latest, v2.

```
yarn add @optimizely/react-sdk@1
```

**Comment by moshie** [comment]
This should be reopened for version 2 as I am now getting the same errors using the .mjs file.

I also tried using the @optimizely/react-sdk/dist/react-sdk.js file but gatsby throws errors when building since there is no window object in a node environment.

<img width="992" alt="Screenshot 2020-09-21 at 09 28 23" src="https://user-images.githubusercontent.com/3974301/93747540-c68ae100-fbee-11ea-8e67-4c143bef6670.png">

**Comment by jderochervlk** [comment]
I tried moving the Optimzely provider out of the `gatsby-ssr` and `gatsby-browser`, but it still fails to build with many of the following type of errors:

```js

 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'useEffect' from non EcmaScript module (only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs


 ERROR #98123  WEBPACK

Generating SSR bundle failed

Can't import the named export 'useState' from non EcmaScript module
(only default export is available)

File: node_modules/@optimizely/react-sdk/dist/react-sdk.mjs
```

**Comment by moshie** [comment]
https://github.com/optimizely/react-sdk/pull/82 FYI

**Comment by Danny-Driscoll** [comment]
Belatedly closing this one out based on fix from #82
